pkg/auth/handler/sso/auth_handler.go

```diff
@@ -153,8 +154,8 @@ func (h AuthHandler) Handle(w http.ResponseWriter, r *http.Request) (success boo
 	}
 
 	// Extract API Key from state
-	key := h.APIClientConfigurationProvider.GetAccessKeyByClientID(state.APIClientID)
-	h.AuthContextSetter.SetAccessKey(key)
+	client, _ := model.GetClientConfig(h.TenantConfiguration.AppConfig.Clients, state.APIClientID)
+	r = r.WithContext(coreAuth.WithAccessKey(r.Context(), coreAuth.AccessKey{Client: client}))
```

Comment:
I guess this is problematic because the `ctx` received by the dependencies are the context WITHOUT access key. There is no way to update the `ctx` of the dependencies.

Reply:
Yes, this is problematic. Originally I store a plain `AccessKey` in context. To workaround this problem for now, a pointer is stored in context to act as a value container, so that `WithAccessKey` can update the existing context.

I planned to refactor it (after authentication session), so that access key in context is only written by middlewares, and read by handlers. Providers should accept the access key (i.e. OAuthClientConfiguration) explicitly through parameters.

Reply:
Also, the access key container middleware is put in all handler (same as authn middleware), and I plan to make it per-handler as well.